### PR TITLE
QBL-2136:ProfilePlus regression test case for connect flow

### PR DESCRIPTION
**Pull request by mridula18**

### CODE REVIEW

File: src/main/java/com/lithium/mineraloil/lia/ProfilePlus/FacebookLogin.java [hunk 1]
```diff
@@ -1,12 +1,30 @@
...
 public class FacebookLogin {
 	private FacebookConnectView facebookConnectView = new FacebookConnectView();
 
 	public void loginToFacebook(String username,String password){
-		facebookConnectView.getFacebookUsername().type(username);
-		facebookConnectView.getFacebookPassword().type(password);
-		facebookConnectView.getFacebookLoginButton().click();
+		if (facebookConnectView.getFacebookUsername().isDisplayed()){facebookConnectView.getFacebookUsername().type(username);}
+		if (facebookConnectView.getFacebookPassword().isDisplayed()){facebookConnectView.getFacebookPassword().type(password);}
+		if (facebookConnectView.getFacebookLoginButton().isDisplayed()){facebookConnectView.getFacebookLoginButton().click();}
```
Comment (JGarcia725) [inline src/main/java/com/lithium/mineraloil/lia/ProfilePlus/FacebookLogin.java]: would one of these checks imply that all are valid? If you can validate the username field, and that meant the rest were present the redundant checks wouldn't be necessary and the code would be cleaner.

File: src/main/java/com/lithium/mineraloil/lia/ProfilePlus/FacebookLogin.java [hunk 2]
```diff
@@ -1,12 +1,30 @@
...
+			}
+		}.setTimeout(TimeUnit.SECONDS, 30).waitAndIgnoreExceptions();
+		if (facebookConnectView.getFacebookContinueButton().isDisplayed())facebookConnectView.getFacebookContinueButton().click();
+		new WaitCondition() {
+			@Override
+			public boolean isSatisfied() {
+				return facebookConnectView.getFacebookConfirmButton().isDisplayed();
+			}
+		}.setTimeout(TimeUnit.SECONDS, 30).waitAndIgnoreExceptions();
+		if (facebookConnectView.getFacebookConfirmButton().isDisplayed())facebookConnectView.getFacebookConfirmButton().click();
```
Comment (JGarcia725) [inline src/main/java/com/lithium/mineraloil/lia/ProfilePlus/FacebookLogin.java]: this "if" check and the one above the wait condition aren't necessary. If you're waiting until these buttons are displayed, you don't need to make sure they're displayed after that condition has been satisfied.

File: src/main/java/com/lithium/mineraloil/lia/ProfilePlus/ProfilePlusSettings.java [hunk 3]
```diff
@@ -1,84 +1,113 @@
...
 	}
+	public void disconnectFacebookfromKlout(){
+		kloutView.open();
+		new WaitCondition() {
+			@Override
+			public boolean isSatisfied() {
+				return kloutView.getKloutLoginSM(SocialMediaNetworks.FACEBOOK.getName()).isDisplayed();
+			}
+		}.setTimeout(TimeUnit.SECONDS, 30).waitAndIgnoreExceptions();
+		if(kloutView.getKloutLoginSM(SocialMediaNetworks.FACEBOOK.getName()).isDisplayed()){
```
Comment (JGarcia725) [inline src/main/java/com/lithium/mineraloil/lia/ProfilePlus/ProfilePlusSettings.java]: this condition should already be met by the wait condition, so this check is unnecessary.

File: src/main/java/com/lithium/mineraloil/lia/ProfilePlus/TwitterLogin.java [hunk 4]
```diff
@@ -4,8 +4,8 @@
 	private TwitterConnectView twitterConnectView = new TwitterConnectView();
 
 	public void loginTotwitter(String username,String password){
-		twitterConnectView.getTwitterUsername().type(username);
-		twitterConnectView.getTwitterPassword().type(password);
-		twitterConnectView.getTwitterLoginButton().click();
+		if (twitterConnectView.getTwitterUsername().isDisplayed()){twitterConnectView.getTwitterUsername().type(username);}
+		if (twitterConnectView.getTwitterPassword().isDisplayed()){twitterConnectView.getTwitterPassword().type(password);}
+		if (twitterConnectView.getTwitterLoginButton().isDisplayed()){twitterConnectView.getTwitterLoginButton().click();}
```
Comment (JGarcia725) [inline src/main/java/com/lithium/mineraloil/lia/ProfilePlus/TwitterLogin.java]: I think one check to see if one of these fields is present should imply the rest are (same case with the facebook checks).

File: src/main/java/com/lithium/mineraloil/lia/ProfilePlus/ProfilePlusSettings.java [hunk 5]
```diff
@@ -67,7 +74,10 @@ public boolean isSatisfied() {
 		kloutView.getKloutDisconnectSM(SocialMediaNetworks.FACEBOOK.getName()).click();
 		kloutView.getKloutDisconnectSMConf().click();
 		kloutView.getKloutDisconnectSMConfirm().click();
-		kloutView.getKloutLogout().click();
+		while(kloutView.getKloutLogout().isDisplayed() && counter<=5){
+			kloutView.getKloutLogout().click();
+			counter++;
```
Comment (JGarcia-li) [inline src/main/java/com/lithium/mineraloil/lia/ProfilePlus/ProfilePlusSettings.java]: Will a refresh help this?

Comment (mridula18) [inline src/main/java/com/lithium/mineraloil/lia/ProfilePlus/ProfilePlusSettings.java]: The refresh is was not  logging me out, i had to click signout several times.

File: src/main/java/com/lithium/mineraloil/lia/ProfilePlus/ProfilePlusSettings.java (hunk 3 above)
Comment (mridula18) [inline src/main/java/com/lithium/mineraloil/lia/ProfilePlus/ProfilePlusSettings.java]: fixed

File: src/main/java/com/lithium/mineraloil/lia/ProfilePlus/TwitterLogin.java (hunk 4 above)
Comment (mridula18) [inline src/main/java/com/lithium/mineraloil/lia/ProfilePlus/TwitterLogin.java]: fixed

**Pull request merged**